README.md

```diff
@@ -63,13 +68,14 @@ Specifically, when an allowed user executes `set_user('rolename')` or
 * [Post-execution hook](#post_set_user_hook)  for `set_user` is called if it is set.
 
 Only users with `EXECUTE` permission on `set_user_u('rolename')` may escalate to
-superuser. Additionally, only roles explicitly listed or included by a group
-that is explicitly listed (e.g. `'+admin'`) in `set_user.superuser_whitelist`
-can escalate to superuser. If `set_user.superuser_whitelist` is explicitly set
-to the empty set, `''`, superuser escalation is blocked for all users. If the
-whitelist is equal to the wildcard character, `'*'`, all users with `EXECUTE`
-permission on `set_user_u()` can escalate to superuser. The default value of
-`set_user.superuser_whitelist` is `'*'`.
+superuser. Additionally, all rules in [whitelist](#whitelist-rules-and-logic)
+apply to `set_user.superuser_whitelist` and `set_user_u('rolename')` with respect
+to the `rolename` argument.
+
+User's calling `set_user('rolename')` can only transition to roles listed or
+included in `set_user.dest_role_whitelist` (defaults to all roles). Additionally
+the [Whitelist logic](#whitelist-rules-and-logic) applies to `current_user` when
+`set_user()` is invoked.
 
```

Comment:
Since we're going to revise this paragraph, let's go with this:

> The following rules govern escalation to superuser via the `set_user_u('rolename')` function:
>
> * `current_user` must be `GRANT`ed `EXECUTE ON FUNCTION set_user_u('rolename')` OR `current_user` must be the OWNER of the `set_user_u` function OR `current_user` must be a superuser.
> * `current_user` must be listed in `set_user.superuser_whitelist` OR `current_user` must belong to a group that is listed in `set_user.superuser_whitelist` (e.g. `'+admin'`)
> * If `set_user.superuser_whitelist` is the empty set , `''`, superuser escalation is blocked for all users.
> * If `set_user.superuser_whitelist` is the wildcard character, `'*'`, all users with `EXECUTE` permission on `set_user_u('rolename')` can escalate to superuser.
> * If `set_user.superuser_whitelist` is not specified, the value defaults to the wildcard character, `'*'`.
>
> The following rules govern non-superuser role transitions through use of the `set_user('rolename')` function:
>
> * `current_user` must be `GRANT`ed `EXECUTE ON FUNCTION set_user('rolename')` OR `current_user` must be the OWNER of the `set_user` function OR `current_user` must be a superuser.
> * The destination `rolename` must be listed in `set_user.dest_role_whitelist` OR the destination `rolename` must belong to a group that is listed in `set_user.dest_role_whitelist` (e.g. `'+client'`)
> * If `set_user.dest_role_whitelist` is the empty set , `''`, `set_user` transitioning to non-superusers is blocked for all users.
> * If `set_user.dest_role_whitelist` is the wildcard character, `'*'`, all users with `EXECUTE` permission on `set_user('rolename')` can transition to any other non-superuser role.
> * If `set_user.dest_user_whitelist` is not specified, the value defaults to the wildcard character, `'*'`.

Reply:
Alright, I'll make these changes, but I'm going to wait on the feedback from above since if we are going to change the names, I'd rather only have to edit this section once.

Reply:
The differences between these two paragraphs illustrate the ambiguity I referenced above. I think I'd prefer to leave the logic explanations split out, since we can be more specific about what the whitelist does for each feature.

Reply:
Oh sorry misunderstanding here, I will keep it split. I meant if I re-write it now I would have to edit it again later if we rename `dest_user_whitelist` and `superuser_whitelist`. So, I want to add that whole section pretty much as is and make the variable name changes all in one go.

Reply:
`as is` == `as you wrote it above`